enterprise/server/raft/driver/driver.go

```diff
@@ -204,21 +214,39 @@ type attemptRecord struct {
 	nextAttemptTime time.Time
 }
 
+type TaskType int
+
+const (
+	_ TaskType = iota
+	RangeTask
+	PartitionTask
+)
+
+type taskKey struct {
+	taskType    TaskType
+	rangeID     uint64 // used for range tasks
+	partitionID string // used for partition tasks
+}
+
 type driverTask struct {
-	rangeID   uint64
-	replicaID uint64
+	key  taskKey
+	repl IReplica // only used for range tasks
+
+	// only used for partition tasks
+	partitionConfig disk.Partition
+	curPD           *rfpb.PartitionDescriptor
```

Comment:
rather than leaving these all at the top level and relying on everyone being well behaved and reading your comment, you could nest them in sub messages that are only set for each task type